### PR TITLE
Implement profile back button

### PR DESCRIPTION
**Pull request by sean-brydon**
## What does this PR do?

Implement back button to take you back to users profile. 

This has been enabled without the history check - Since we migrated over to get static props for this page - tacking the users previous page is difficult and we don't have access to `document.referer` so can't get it the previous way. 

**Environment**: Staging(main branch) / Production

## Type of change

- [X] Bug fix (non-breaking change which fixes an issue)

## Checklist

- I haven't added tests that prove my fix is effective or that my feature works

### CODE REVIEW

File: apps/web/components/booking/pages/AvailabilityPage.tsx [hunk 1]
```diff
@@ -550,15 +551,23 @@ const AvailabilityPage = ({ profile, eventType }: Props) => {
...
                           title: user.name,
                           alt: user.name,
                           image: `${CAL_URL}/${user.username}/avatar.png`,
                         })),
-                    ].filter((item) => !!item.image) as { image: string; alt?: string; title?: string }[]
+                    ].filter((item) => !!item.image) as {
+                      image: string;
+                      alt?: string;
+                      title?: string;
+                    }[]
```
Comment (zomars) [inline apps/web/components/booking/pages/AvailabilityPage.tsx:570]: Unnecessary diff

File: apps/web/components/booking/pages/AvailabilityPage.tsx [hunk 2]
```diff
@@ -550,15 +551,23 @@ const AvailabilityPage = ({ profile, eventType }: Props) => {
                   border="border-2 dark:border-gray-800 border-white"
                   items={
                     [
-                      { image: profile.image, alt: profile.name, title: profile.name },
+                      {
+                        image: profile.image,
+                        alt: profile.name,
+                        title: profile.name,
+                      },
                       ...eventType.users
```
Comment (zomars) [inline apps/web/components/booking/pages/AvailabilityPage.tsx:559]: Unnecessary diff

File: apps/web/components/booking/pages/AvailabilityPage.tsx [hunk 3]
```diff
@@ -411,15 +404,23 @@ const AvailabilityPage = ({ profile, eventType }: Props) => {
...
                           title: user.name,
                           image: `${CAL_URL}/${user.username}/avatar.png`,
                           alt: user.name || undefined,
                         })),
-                    ].filter((item) => !!item.image) as { image: string; alt?: string; title?: string }[]
+                    ].filter((item) => !!item.image) as {
+                      image: string;
+                      alt?: string;
+                      title?: string;
+                    }[]
```
Comment (zomars) [inline apps/web/components/booking/pages/AvailabilityPage.tsx:423]: Unnecessary diff

File: apps/web/components/booking/pages/AvailabilityPage.tsx [hunk 4]
```diff
@@ -411,15 +404,23 @@ const AvailabilityPage = ({ profile, eventType }: Props) => {
                   border="border-2 dark:border-gray-800 border-white"
                   items={
                     [
-                      { image: profile.image, alt: profile.name, title: profile.name },
+                      {
+                        image: profile.image,
+                        alt: profile.name,
+                        title: profile.name,
+                      },
```
Comment (zomars) [inline apps/web/components/booking/pages/AvailabilityPage.tsx:411]: Unnecessary diff

File: apps/web/components/booking/pages/AvailabilityPage.tsx [hunk 5]
```diff
@@ -95,23 +96,15 @@ export const locationKeyToString = (location: LocationObject, t: TFunction) => {
...
-
-  return previousPage === `${WEBAPP_URL}/${slug}` ? (
+  return (
     <div className="flex h-full flex-col justify-end">
-      <ArrowLeftIcon
-        className="h-4 w-4 text-black transition-opacity hover:cursor-pointer dark:text-white"
-        onClick={() => router.back()}
-      />
-      <p className="sr-only">Go Back</p>
+      <button title={t("profile")} onClick={() => router.push(`${WEBSITE_URL}/${slug}`)}>
```
Comment (zomars) [inline apps/web/components/booking/pages/AvailabilityPage.tsx:103]: Do we want to use `push` for "Go back" functionality? If I press this button and then press back in the browser I will be taken back to the availability page. Is this intentional or unexpected?

**Review by zomars — DISMISSED** [review]

**Comment by Jaibles** [comment]
@sean-brydon back to the users profile from where? I checked from availability as that seems to be where the change in the code is but I don't see a back button there when checking out the preview url.

**Comment by sean-brydon** [comment]
@Jaibles Back button is in the bottom left

image

File: apps/web/components/booking/pages/AvailabilityPage.tsx [hunk 6]
```diff
@@ -87,23 +88,15 @@ export const locationKeyToString = (location: LocationObject, t: TFunction) => {
...
+  return (
     <div className="flex h-full flex-col justify-end">
-      <ArrowLeftIcon
-        className="h-4 w-4 text-black transition-opacity hover:cursor-pointer dark:text-white"
-        onClick={() => router.back()}
-      />
-      <p className="sr-only">Go Back</p>
+      <button title={t("profile")} onClick={() => router.replace(`${WEBSITE_URL}/${slug}`)}>
+        <ArrowLeftIcon className="h-4 w-4 text-black transition-opacity hover:cursor-pointer dark:text-white" />
+        <p className="sr-only">Go Back</p>
```
Comment (emrysal) [inline apps/web/components/booking/pages/AvailabilityPage.tsx:97]: Should this be translated?

Comment (sean-brydon) [reply apps/web/components/booking/pages/AvailabilityPage.tsx:97]: Yeah I think it should.

However, thinking about it I'm not sure if we need it for a11y standards due to the fact we are already using the title attribute.

**Review by emrysal — APPROVED** [review]
Looks good to me, translation issue is up to you

**Review by zomars — APPROVED** [review]
